### PR TITLE
core,test-util: fix type inference of createRouteRef + add mountedRoutes option to wrapInTestApp

### PR DESCRIPTION
**Pull request by Rugvip**
## Hey, I just made a Pull Request!

`createRouteRef` wasn't properly inferring the specific list of parameters, or an empty list of parameters, which are the two things it was supposed to do 😅 

Adding the `mountedRoutes` to `wrapInTestApp`, and `renderInTestApp`, fills in a gap were there's currently no sane way to test components that use `useRouteRef` except for mocking the hook.

It's used like this:

```tsx
const aRouteRef = createRouteRef({ title: 'A' });
const bRouteRef = createRouteRef({ title: 'B', params: ['name'] });

const MyComponent = () => {
  const a = useRouteRef(aRouteRef);
  const b = useRouteRef(bRouteRef);
  return (
    <div>
      <div>Link A: {a()}</div>
      <div>Link B: {b({ name: 'x' })}</div>
    </div>
  );
};

const rendered = await renderInTestApp(<MyComponent />, {
  mountedRoutes: {
    '/my-a-path': aRouteRef,
    '/my-b-path/:name': bRouteRef,
  },
});
expect(rendered.getByText('Link A: /my-a-path')).toBeInTheDocument();
expect(rendered.getByText('Link B: /my-b-path/x')).toBeInTheDocument();
```

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)

### CODE REVIEW

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4407-w85r0gffd95l5mmbsyozpipor3a9bcsk.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/601d799f14dc586552c42d8c

**Comment by Fox32** [comment]
Thaaaannnnkkkkss! That's what I was looking for!

File: packages/core-api/src/routing/RouteRef.ts [hunk 1]
```diff
@@ -38,9 +47,15 @@ export class AbsoluteRouteRef<Params extends { [param in string]: string }> {
 }
 
 export function createRouteRef<
-  ParamKeys extends string,
-  Params extends { [param in string]: string } = { [name in ParamKeys]: string }
->(config: RouteRefConfig<Params>): RouteRef<Params> {
+  Params extends { [param in ParamKey]: string },
+  ParamKey extends string = never
```
Comment (freben) [inline packages/core-api/src/routing/RouteRef.ts:51]: I don't get this one frankly :)

**Review by freben — APPROVED** [review]

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4407-w85r0gffd95l5mmbsyozpipor3a9bcsk.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/601d799f14dc586552c42d8c

**Pull request merged**